docs/glossary.md

```diff
@@ -0,0 +1,161 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a branch that doesn't have any [branches](#branch)
+pointing to it or to any of its descendants. Unlike Git, Jujutsu keeps such
+commits around until they are explicitly abandoned.
+
+## Backend
+
+A backend is an implementation of the storage layer. When unqualified, we
+typically mean "commit backend", which is the storage layer for commits. There
+are also pluggable backends for e.g. storing [the operation log](#operation-log)
+.
+
+## Branch
```

Comment:
I would call this "Named branch" and start the description with `A named branch, often referred to simply as a "branch"...`

Reply:
I think we almost always call them just "branches". I added a note about "named branches" instead. Let me know what you think.

Reply:
Seems fine.